modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/EitherTTest.kt

```diff
@@ -46,12 +49,22 @@ class EitherTTest : UnitSpec() {
     a.value().attempt().unsafeRunTimed(60.seconds) == b.value().attempt().unsafeRunTimed(60.seconds)
   }
 
+  fun GEN() =
+    Gen.constant(
+      EitherT.applicative<ForId, Int>(Id.monad()))
+
   init {
 
+    val cf: (Int) -> EitherT<Kind<ForConst, Int>, Int, Int> = { EitherT(it.const()) }
+    val g = Gen.int().map(cf) as Gen<Kind<EitherTPartialOf<ConstPartialOf<Int>, Int>, Int>>
+
+    val cfId: (Int) -> EitherT<ForId, Nothing, Int> = { EitherT(Id(Right(it))) }
+    val genId = Gen.intSmall().map(cfId) as Gen<Kind<EitherTPartialOf<ForId, Int>, Int>>
```

Comment:
You can actually use `GenK` here, with `GenK<EitherTPartialOf<F, L>>` or something similar. You could define this by taking a `GenK<F>` and a `Gen<L>` as parameters to test all cases.

Reply:
done. it was good to do the Sum EqK before.